covidthailand.py

```diff
@@ -3467,156 +3876,208 @@ def save_plots(df):
     #ax.get_yaxis().get_major_formatter().set_scientific(False)
-    #cols = ["Vaccinations Given 1 Cum", "Vaccinations Given 2 Cum"]
-    cols = [c for c in df.columns if str(c).startswith("Vac Group")]
-    leg = lambda c: c.replace(" Cum","").replace("Vac Group","").replace("1", "Dose 1").replace("2", "Dose 2")
+    #cols = ['Vaccinations Given 1 Cum', 'Vaccinations Given 2 Cum']
+    cols = [c for c in df.columns if str(c).startswith('Vac Group')]
+    leg = lambda c: c.replace(' Cum','').replace('Vac Group','').replace('1', 'Dose 1').replace('2', 'Dose 2')
     cols.sort(key=lambda c: leg(c)[-1]+leg(c)) # put 2nd shot at end
     # some missing data. should be able to fill it in
-    df["2021-02-16":][cols].interpolate().plot.area(
-        ax=ax,
-        y=cols,
-        colormap="Set3",
-        title='Thailand Vaccinations by Groups\n'
-        "(% of 2 doses per Thai population)\n"
-        f"Updated: {TODAY().date()}\n"        
-        "djay.github.io/covidthailand"
-    )
-    ax.legend([leg(c) for c in cols])
-    plt.tight_layout()
-    plt.savefig("outputs/vac_groups_3.png")
-
-    cols = rearrange([f"Vac Given 1 Area {area} Cum" for area in range(1, 14)],*FIRST_AREAS)
-    fig, ax = plt.subplots(figsize=[20, 10],)
-    ax.yaxis.set_major_formatter(FuncFormatter(thaipop))
-    df["2021-02-16":][cols].interpolate().plot.area(
-        ax=ax,
-        y=cols,
-        colormap=AREA_COLOURS,
-        title='Thailand Vaccinations (1st Shot) by Health District\n'
-        "(% per population)\n"
-        f"Updated: {TODAY().date()}\n"        
-        "djay.github.io/covidthailand"
-    )
-    ax.legend(AREA_LEGEND)
-    plt.tight_layout()
-    plt.savefig("outputs/vac_areas_s1_3.png")
-
-    cols = rearrange([f"Vac Given 2 Area {area} Cum" for area in range(1, 14)],*FIRST_AREAS)
-    fig, ax = plt.subplots(figsize=[20, 10],)
-    ax.yaxis.set_major_formatter(FuncFormatter(thaipop))
-    df["2021-02-16":][cols].interpolate().plot.area(
-        ax=ax,
-        y=cols,
-        colormap=AREA_COLOURS,
-        title='Thailand Fully Vaccinatated (2nd Shot) by Health District\n'
-        "(% population full vaccinated)\n"
-        f"Updated: {TODAY().date()}\n"        
-        "djay.github.io/covidthailand"
-    )
-    ax.legend(AREA_LEGEND)
-    plt.tight_layout()
-    plt.savefig("outputs/vac_areas_s2_3.png")
+    # df['2021-02-16':][cols].interpolate().plot.area(
+    #     ax=ax,
+    #     y=cols,
+    #     colormap='Set3',
+    #     title='Thailand Vaccinations by Groups\n'
+    #     '(% of 2 doses per Thai population)\n'
+    #     f'Updated: {TODAY().date()}\n'
+    #     'djay.github.io/covidthailand'
+    # )
+    # ax.legend([leg(c) for c in cols])
+    # plt.tight_layout()
+    # plt.savefig('outputs/vac_groups_3.png')
+    legends = [leg(c) for c in cols]
+    df_vac_groups_3 = df['2021-02-16':][cols].interpolate()
+    plot_area(df=df_vac_groups_3, png_prefix='vac_groups_3', cols_subset=cols,
```

Comment:
_3 in the name means the time period so shouldn't be in the prefix.

Reply:
Ok

Reply:
Ok